src/workflows/orders.ts

```diff
@@ -104,3 +168,45 @@ export async function cancelOrderWorkflow(
     status: cancelOrderResponse.data.status,
   };
 }
+
+export async function cancelOrderWorkflowWithSigner({
+  l1Signer,
+  l2Signer,
+  request,
+  ordersApi,
+}: CancelOrderWorkflowWithSignerRequest) {
+  const getSignableCancelOrderResponse = await ordersApi.getSignableCancelOrder(
+    {
+      getSignableCancelOrderRequest: {
+        order_id: request.order_id,
+      },
+    },
+  );
+
+  const { signable_message: signableMessage, payload_hash: payloadHash } =
+    getSignableCancelOrderResponse.data;
+
+  // Sign message with L1 credentials
+  const ethSignature = await signRaw(signableMessage, l1Signer);
+
+  // Sign hash with L2 credentials
+  const starkSignature = await l2Signer.signMessage(payloadHash);
+
+  // Obtain Ethereum Address from signer
+  const ethAddress = await l1Signer.getAddress();
+
+  const cancelOrderResponse = await ordersApi.cancelOrder({
+    id: request.order_id.toString(),
+    cancelOrderRequest: {
+      order_id: request.order_id,
+      stark_signature: starkSignature,
+    },
+    xImxEthAddress: ethAddress,
+    xImxEthSignature: ethSignature,
+  });
+
+  return {
+    order_id: cancelOrderResponse.data.order_id,
```

Comment:
why we just not return {...cancelOrderResponse.data}?

Reply:
Return this is because that's what `cancelOrderWorkflow` returns

The function `cancelOrderWorkflowWithSigner` is based on the existed in implementation, `cancelOrderWorkflow`. the difference is we changed the input parameters to accept the l2Signer and sign the message from the l2Signer. For the rest of the code, I was directly copied from the existed function to keep the output the same.